### ISSUE TITLE
Automatically suggest useful type aliases

### ISSUE THREAD

**Issue by samth** [racket/typed-racket#334]
When a type is parsed, if it's equal to an existing type alias, suggest that, perhaps as a tooltip.

Suggested by @frenchy64 and inspired by an ESOP paper about macro driven refactoring.

**Comment by jsmaniac** [comment]
When the alias can't be written because it has different scopes than the full type, should it still be suggested?

For example, the user can't refactor `(List String Number)` into `Foo` below, because the `Foo` identifier is introduced by a hygienic macro.

```
#lang typed/racket

(define-syntax-rule (def-hidden-type)
  (define-type Foo (List String Number)))

(def-hidden-type)

(ann '("a" 2) (List String Number))
```

Apart from confusion, I don't see any other problem caused by still having the tooltip even if the refactoring can't be done, though.

**Comment by lexi-lambda** [comment]
I agree with @jsmaniac’s concerns here, though I also think this sounds like it would be a very neat feature. It seems like some of the work that would need to be done to address those concerns might also be able to improve the issues with how types are currently printed (type aliases are global, sometimes types not declared using an alias print using that alias, sometimes types declared using an alias *don’t* print under that alias, etc.). Of course, on the other hand, that might take a lot more work because the way aliases are represented might need to be pretty significantly changed.

Still, the reason I bring this up is just because of the weird behavior I’ve seen with aliases in the past. Since these are just suggestions not visible by default, it’s probably ok, though.
